src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -77,5 +87,87 @@ static Attribute ProcessRequiresUnreferencedCodeAttribute (LinkContext context,
 				2028, MessageOrigin.TryGetOrigin (method, 0));
 			return null;
 		}
+
+		static DynamicDependency ProcessDynamicDependencyAttribute (LinkContext context, ICustomAttributeProvider provider, CustomAttribute customAttribute)
+		{
+			if (!(provider is MethodDefinition || provider is FieldDefinition))
+				return null;
+			var member = provider as IMemberDefinition;
+
+			if (!ShouldProcessDependencyAttribute (context, customAttribute))
+				return null;
+
+			var dynamicDependency = GetDynamicDependency (context, customAttribute);
+			if (dynamicDependency != null)
+				return dynamicDependency;
+
+			context.LogMessage (MessageContainer.CreateWarningMessage (context,
+				$"Invalid DynamicDependencyAttribute on '{member}'",
+				2034, MessageOrigin.TryGetOrigin (member)));
+			return null;
+		}
+
+		static DynamicDependency GetDynamicDependency (LinkContext context, CustomAttribute ca)
```

Comment:
Isn't this duplicated between DynamicDependency and here? I would keep the one in DynamiDependency.

Reply:
Good catch, thank you.